libevm/VM.cpp

```diff
@@ -573,6 +573,47 @@ void VM::interpretCases()
 		}
 		NEXT
 
+#ifdef EVM_USE_BITSHIFT
+		CASE(SHL)
+		{
+			ON_OP();
+			updateIOGas();
+
+			/// TODO: confirm shift >= 256 results in 0
```

Comment:
That is done in #4064. So we should add appropriate boundary checks here (and masking).